platforms/core-runtime/base-services/src/main/java/org/gradle/internal/work/DefaultWorkerLeaseService.java

```diff
@@ -45,27 +41,31 @@
 import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 
 import static org.gradle.internal.resources.DefaultResourceLockCoordinationService.lock;
 import static org.gradle.internal.resources.DefaultResourceLockCoordinationService.tryLock;
 import static org.gradle.internal.resources.DefaultResourceLockCoordinationService.unlock;
 
 public class DefaultWorkerLeaseService implements WorkerLeaseService, ProjectParallelExecutionController, Stoppable {
-    public static final String PROJECT_LOCK_STATS_PROPERTY = "org.gradle.internal.project.lock.stats";
+
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultWorkerLeaseService.class);
 
     private final WorkerLimits workerLimits;
     private final ResourceLockCoordinationService coordinationService;
     private final WorkerLeaseLockRegistry workerLeaseLockRegistry;
-    private final ProjectLockStatisticsImpl projectLockStatistics = new ProjectLockStatisticsImpl();
+    private final ResourceLockStatistics resourceLockStatistics;
     private final AtomicReference<Registries> registries = new AtomicReference<Registries>(new NoRegistries());
 
-    public DefaultWorkerLeaseService(ResourceLockCoordinationService coordinationService, WorkerLimits workerLimits) {
+    public DefaultWorkerLeaseService(
+        ResourceLockCoordinationService coordinationService,
+        WorkerLimits workerLimits,
+        ResourceLockStatistics resourceLockStatistics
```

Comment:
💅 Since this class is already coupled to `ResourceLockStatistics` it could have another constructor for a default `NO_OP` value and avoid the cascading changes on every client.